src/olympia/promoted/tasks.py

```diff
@@ -106,3 +109,65 @@ def on_stripe_charge_failed(event):
         from_email=settings.ADDONS_EMAIL,
         recipient_list=["amo-admins@mozilla.com"],
     )
+
+
+@task
+def on_stripe_customer_subscription_deleted(event):
+    event_id = event.get("id")
+    event_type = event.get("type")
+
+    if event_type != "customer.subscription.deleted":
+        log.error(
+            'invalid event "%s" received (event_id=%s).', event_type, event_id
+        )
+        return
+
+    # This event should contain a `subscription` object.
+    subscription = event.get("data", {}).get("object")
+    if not subscription:
+        log.error("no subscription object in event (event_id=%s).", event_id)
+        return
+
+    subscription_id = subscription["id"]
+    log.info(
+        'received "%s" event with subscription_id=%s.',
+        event_type,
+        subscription_id,
+    )
+
+    try:
+        sub = PromotedSubscription.objects.get(
+            stripe_subscription_id=subscription_id
+        )
+    except PromotedSubscription.DoesNotExist:
+        log.error(
+            'received a "%s" event (event_id=%s) for a non-existent'
+            " promoted subscription (subscription_id=%s).",
+            event_type,
+            event_id,
+            subscription_id,
+        )
+        return
+
+    if sub.cancelled_at:
+        log.info(
+            "PromotedSubscription %s already cancelled, ignoring event"
+            " (event_id=%s).",
+            sub.id,
+            event_id,
+        )
+        return
+
+    sub.update(
+        cancelled_at=datetime.datetime.fromtimestamp(
+            subscription["canceled_at"]
+        )
+    )
+    log.info("PromotedSubscription %s has been marked as cancelled.", sub.id)
+
+    # TODO: this should trigger resigning somehow, see:
```

Comment:
@eviljeff FYI but also a question: should we try to land https://github.com/mozilla/addons-server/issues/15921 first?

Reply:
It doesn't seem a strict prerequisite so I'd say no.